projects/ngx-d3timeline/src/lib/hover/hover-event.ts

```diff
@@ -15,15 +15,15 @@ export interface HoverEvent {
   action: HoverAction;
 }
 
-export function getHoveredActivityByType(
+export function getHoveredActivity(
   hoverEvent: HoverEvent,
-  hoveredPositionedActivity: PositionedActivity,
-  action: HoverAction
+  hoveredPositionedActivity: PositionedActivity
 ): Activity {
   return (
-    hoverEvent &&
-    (hoverEvent.action === action
-      ? getActivityFromPositionedActivity(hoveredPositionedActivity)
-      : null)
+    hoverEvent && getActivityFromPositionedActivity(hoveredPositionedActivity)
   );
 }
+
+export function hoverEventComparator(x: HoverEvent, y: HoverEvent) {
```

Comment:
I think better not to use x and y as it implies some form of coordinates/position.

```suggestion
export function hoverEventComparer(hoverEvent1: HoverEvent, hoverEvent2: HoverEvent) {
```

Reply:
I prefer using single letter variables in these sort of functions.  I think here having to parse the longer variable names make the code harder to read.